frontend/user/routes.py

```diff
@@ -15,10 +16,22 @@
 @user_bp.route("")
 @login_requested
 def user():
+    roles_required = request.args.get("roles_required")
     logged_in_user = g.user if g.is_authenticated else None
+    roles_error = False
+    if logged_in_user and (
+        not logged_in_user.roles
```

Comment:
I wonder, if we only show a roles_required error when the user is logged in, perhaps all the roles required logic should go inside an `if logged_in_user:` block?

Reply:
we could then just lose the roles_error flag and instead just validate to check that the roles_required are actually required eg.
```suggestion
        if logged_in_user:
             if roles_required and logged_in_user.roles 
             and all(role_required in logged_in_user.roles
            for role_required in roles_required):
                 roles_required = None
```
...and then just pass roles_required to the template and build the logic off that...? Any thoughts?

Reply:
edited with required changes